code/modules/hazmat/crystal_maker.dm

```diff
@@ -0,0 +1,20 @@
+/obj/machinery/hazmat
+	icon = 'icons/obj/hazmat/machinery.dmi'
+
+/obj/machinery/hazmat/crystal_maker
+	name = "crystal maker"
+	desc = "Insert diamonds or glass to generate crystals. Uses 1 sheet each."
+	icon_state = "crystal_maker"
+
+/obj/machinery/hazmat/crystal_maker/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I, /obj/item/stack/sheet/glass))
```

Comment:
this code is snowflakey and should probably be handled through datum mats or atleast not copypaste the same crystal spawnign behavior. Make a MakeCrystal proc or otherwise handle this a bit less copy paste

Reply:
ill check it out but no guarantees, current method works

Reply:
we have a system set-up for materials, hacking around it by adding snowflake checks with copy pasted code is not something you should do.